Directory.Packages.props

```diff
@@ -14,6 +14,8 @@
     </PropertyGroup>
 
     <ItemGroup>
+        <PackageVersion Include="AsyncKeyedLock" Version="6.2.4" />
```

Comment:
Currently the code allows concurrent calls to `RestoreAsync` for the same `workingDirectory` value. I believe this could lead to undesired results so it is safer to stop concurrent processing for the same working directory, which is what this PR achieves.